### PR TITLE
feat/monorepo

### PR DESCRIPTION
**Pull request by cdaringe**
# problem

- there are few packages cram-jammed together in the current state, as though they were one
- we probably want to mono it up so we can also checkin a docs site :)

# solution

- lerna
- typescript mono, using `references` -- `npx tsc -b packages`--efficient bulk compile 👌 
- break up the things into three subpackages
  - `effects-common`
  - `effects-runtime`
  - `babel-plugin-effects`

# wip

- [ ] un-destroy dave's readme docs

### CODE REVIEW

File: .github/contributing.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# contributing
+
+Thanks for you interest in contributing.
```
Comment (cdaringe) [inline .github/contributing.md]: Caps ;)

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-12.4.0
+13.8.0
```
Comment (dpchamps) [inline .nvmrc]: Why not stick with 12? Is there any advantage to bumping to 13? I typically shy away from odd-numbered releases unless there's something specific I want from it.

File: package.json [hunk 3]
```diff
@@ -1,40 +1,30 @@
 {
-  "name": "effects-js",
-  "version": "0.0.1",
-  "description": "Algebraic Effects For Javascript 🎉",
-  "main": "lib/index.js",
+  "name": "effectsjs",
+  "version": "1.0.0",
```
Comment (dpchamps) [inline package.json]: 😱

File: .gitignore [hunk 4]
```diff
@@ -30,9 +33,6 @@ build/Release
 # https://www.npmjs.org/doc/misc/npm-faq.html#should-i-check-my-node_modules-folder-into-git
 node_modules
 
-#ccm
-.ccm
-.idea
```
Comment (dpchamps) [inline .gitignore:35]: ~I prefer not to commit IDE folders. Thoughts?~ nvm I see you added it above

File: .github/contributing.md (hunk 1 above)
Comment (dpchamps) [inline .github/contributing.md]:
```suggestion
Thanks for your interest in contributing.
```

**Review by dpchamps — APPROVED** [review]
This is great! Way better structure. Pushing some changes to fix a couple of things..

One major bummerage: code coverage reporting has been thrashed. Something to think about later.

**Comment by dpchamps** [comment]
Thoughts on examples? 

Maybe a PR after this one, we can break examples out into another package to show how everything would be used in a normal projects. Maybe overkill?

File: .gitignore (hunk 4 above)
Comment (cdaringe) [reply .gitignore:35]: i actually do like checking them in, but carrying this over from olden times.  #diversity.  also, nbd who cares

File: .nvmrc (hunk 2 above)
Comment (cdaringe) [reply .nvmrc]: 353a57a

Comment (cdaringe) [reply .nvmrc]: 353a57a

File: package.json (hunk 3 above)
Comment (cdaringe) [reply package.json]: top level mono root--not a real pkg :). doesn't mean anything.  952c0a0 tho

**Comment by cdaringe** [comment]
if we can get your compiler version built in browser, i even want examples live, on the web. runable, editable.

**Comment by dpchamps** [comment]
It's definitely possible. Some guy made a ast viewer that allowed for uploading customer parsers. 

I'll make an issue for it

**Pull request merged**